lib/util/exec/index.ts

```diff
@@ -25,46 +25,48 @@ export interface ExecOptions extends ChildProcessExecOptions {
 }
 
 function getChildEnv({
-  extraEnv = {},
+  extraEnv,
   env: forcedEnv = {},
-}: ExecOptions): ExtraEnv<string> {
+}: ExecOptions): Record<string, string> {
   const globalConfigEnv = GlobalConfig.get('customEnvVariables');
 
-  const inheritedKeys = Object.entries(extraEnv).reduce(
-    (acc, [key, val]) =>
-      val === null || val === undefined ? acc : [...acc, key],
-    []
-  );
+  const inheritedKeys: string[] = [];
+  for (const [key, val] of Object.entries(extraEnv ?? {})) {
+    if (val !== null && val !== undefined) {
+      inheritedKeys.push(key);
+    }
+  }
+
   const parentEnv = getChildProcessEnv(inheritedKeys);
-  const childEnv = Object.entries({
+  const combinedEnv = {
     ...extraEnv,
     ...parentEnv,
     ...globalConfigEnv,
     ...forcedEnv,
-  }).reduce(
-    (acc, [key, val]) =>
-      val === null || val === undefined
-        ? acc
-        : { ...acc, [key]: val.toString() },
-    {}
-  );
-  return childEnv;
+  };
+
+  const result: Record<string, string> = {};
+  for (const [key, val] of Object.entries(combinedEnv)) {
+    if (val !== null && val !== undefined) {
+      result[key] = `${val}`;
+    }
+  }
+
+  return result;
 }
 
-function dockerEnvVars(
-  extraEnv: ExtraEnv,
-  childEnv: ExtraEnv<string>
-): string[] {
+function dockerEnvVars(extraEnv: ExtraEnv, childEnv: ExtraEnv): string[] {
   const extraEnvKeys = Object.keys(extraEnv || {});
-  return extraEnvKeys.filter(
-    (key) => typeof childEnv[key] === 'string' && childEnv[key].length > 0
-  );
+  return extraEnvKeys.filter((key) => {
+    const val = childEnv[key];
+    return typeof val === 'string' && val.length > 0;
+  });
 }
 
-function getCwd({ cwd, cwdFile }: ExecOptions): string {
+function getCwd({ cwd, cwdFile }: ExecOptions): string | undefined {
```

Comment:
I think this can only return `undefined` on tests, we should assume it always returns a string

`GlobalConfig.get('localDir')` should always return the repo root path, `undefined` is an error